jekyll/_cci2/configuration-reference.md

```diff
@@ -517,6 +574,16 @@ steps:
       when: on_fail
 ```
 
+##### **`when`** (requires version: 2.1)
```

Comment:
@michelle-luna Can you remove this section? I think it's out of scope for Cayenne's pull request, and needs a more explanation (a `when` step can go in commands and jobs) and a filled out example.  Feel free to open this as another PR and I'll work with you to create examples and explanation.

Reply:
I'm not sure if it was the right thing to do, but I did move that card to done without waiting for this to be merged.

I think it does make sense to document all the 2.1 stuff at once.

Reply:
Thanks both of you for working on this!

Reply:
@michelle-luna in that case, you can disregard my earlier comment about removing this section from this PR. This section still doesn't quite fit under steps/when. I'll follow up this comment with a couple of examples.

Reply:
@michelle-luna 

Here's an example of using "when" and "unless" in a job:

```yaml
$ circleci config process asdf.yml
version: 2.1
jobs:
  job_with_optional_custom_checkout-1:
    machine: true
    steps:
    - run:
        command: echo "my custom checkout"
  job_with_optional_custom_checkout-2:
    machine: true
    steps:
    - checkout
workflows:
  build-test-deploy:
    jobs:
    - job_with_optional_custom_checkout-1
    - job_with_optional_custom_checkout-2
  version: 2

# Original config.yml file:
# version: 2.1
#
# jobs:
#   job_with_optional_custom_checkout:
#     parameters:
#       custom_checkout:
#         type: string
#         default: \"\"
#     machine: true
#     steps:
#       - when:
#           condition: <<parameters.custom_checkout>>
#           steps:
#             - run: echo \"my custom checkout\"
#       - unless:
#           condition: <<parameters.custom_checkout>>
#           steps:
#             - checkout
#
# version: 2.1
# workflows:
#   build-test-deploy:
#     jobs:
#       - job_with_optional_custom_checkout:
#           custom_checkout: \"any non-empty string is truthy\"
#       - job_with_optional_custom_checkout
```

And an example of using `when` in a command:

```yaml
$ circleci config process asdf.yml
version: 2.1
jobs:
  simple_job:
    machine: true
    steps:
    - checkout
    - run:
        command: echo "my custom checkout"
workflows:
  build-test-deploy:
    jobs:
    - simple_job
  version: 2

# Original config.yml file:
# version: 2.1
#
# commands:
#   my_custom_checkout:
#     parameters:
#       custom_checkout:
#         type: string
#         default: \"\"
#     steps:
#       - when:
#           condition: <<parameters.custom_checkout>>
#           steps:
#             - run: echo \"my custom checkout\"
#       - unless:
#           condition: <<parameters.custom_checkout>>
#           steps:
#             - checkout
#
#
# jobs:
#   simple_job:
#     machine: true
#     steps:
#       - my_custom_checkout
#       - my_custom_checkout:
#           custom_checkout: \"a non-empty string\"
#
# version: 2.1
# workflows:
#   build-test-deploy:
#     jobs:
#       - simple_job
```

This should be in a separate section from the existing `when` that goes inside of `run`.  They take different arguments, are used for different purposes, and users will have to think about data at different points in their build's lifecycle: 
- customizing a step to run on fail, on pass, or always (determined while the build is running)
- customizing commands and job configuration to run on custom conditions (determined at config-compile time)

 The shape is identical between commands and jobs, so their documentation can be shared.

Reply:
Thanks, it is in a different section from the when attribute. I qualified the name of this section as 'The `when` Step' to be more clear. I also added one example (this section will point to the complete section in the Reusing Config document for the additional information about conditional steps.

Reply:
Just checked this section out in the docs. Looks good 👍

Reply:
thanks @eric-hu! I will add in the live links today in conjunction with doing the changelog